docs/proposals/diagnose-experience.md

```diff
@@ -0,0 +1,247 @@
+
+## Problem Statement
+
+Specifically, there are exiting GitHub issues raised in the ORAS community.
+
+- The user is confused about when to use `--verbose` and `--debug`. See the relevant issue [#1382](https://github.com/oras-project/oras/issues/1382).
+- Poor readability of debug logs. No separator lines between request and response information. Users need to add separator lines manually for readability. See the relevant issue [#1382](https://github.com/oras-project/oras/issues/1382).
+- Critical information is missing in debug logs. For example, the [error code](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#error-codes) and metadata of the processed resource object (e.g. image manifest) are not displayed.
+- The detailed operation information is missing in verbose output. For example, how many resource objects are processed where. Less or none verbose output of ORAS commands in some use cases.
+- Timestamp of each request and response is missing in debug logs, which is hard to trace historical operation and trace the sequence of events accurately.
+
+## Concepts
+
+Before restructuring the log information of ORAS debug log and [verbose output](https://en.wikipedia.org/wiki/Verbose_mode), it worth clarifying the concepts and differences between verbose output and debug logs.
+
+### Verbose Output 
+
+Verbose output focuses on providing a comprehensive, high-level view of the application's operations. It is intended for end-users who want to observe the detailed normal operation of the tool. It should be human readable and descriptive.
+
+- **Purpose**: Verbose output are typically used to provide highly detailed information about the flow and operations of the application. They give users a comprehensive view of what the tool is doing at every step.
+- **Target users**: These are generally intended for users who want to understand the detailed workings of the tool, not necessarily for debugging specific issues.
+- **Content**: Verbose output includes lots of informational messages about the application's state, operations performed, configuration details, and more. They provide a broader view, which is helpful for tracing the overall execution of the tool.
+- **Level of Detail**: Very detailed, but usually focused on normal operations rather than errors or issues.
+
+### Debug Logs
+
+Debug logs focus on providing technical details for in-depth diagnosing and troubleshooting issues. It is intended for developers or technical users who need to understand the inner workings of the tool. Debug logs are detailed and technical, often including HTTP request and response from interactions between client and server, as well as code-specific information.
+
+- **Purpose**: Debug logs are specifically aimed at helping developers diagnose and fix issues within the application. They contain detailed technical information that is useful for troubleshooting problems.
```

Comment:
The term application needs context. If its a user, their app might be a script or a GH action making a call to the tool. 
Debug logs in ORAS's case in my opinion should target submitting bug reports to ORAS incase of failed behavior. 
So I would say the goal is to facilitate ORAS developers to fix or diagnose ORAS. Not for a tool user to fix their application.

Reply:
Agree. Updated